packages/interface-adapter/lib/adapter/web3/index.ts

```diff
@@ -53,4 +56,30 @@ export class Web3InterfaceAdapter implements InterfaceAdapter {
   public getBlockNumber() {
     return this.web3.eth.getBlockNumber();
   }
+
+  public async getTransactionReportData(receipt: TransactionReceipt): Promise<TransactionReportData> {
+    const tx = await this.getTransaction(receipt.transactionHash);
+    const balance = await this.getBalance(tx.from);
+
+    const block = await this.getBlock(receipt.blockNumber);
+    if (!block) return null;
+
+    const gasPrice = new BN(tx.gasPrice);
+    const gas = new BN(receipt.gasUsed);
+    const value = new BN(tx.value);
+    const cost = gasPrice.mul(gas).add(value);
+
+    return {
+      timestamp: block.timestamp,
+      from: tx.from,
+      balance: Web3Shim.utils.fromWei(balance, "ether"),
+      gasPrice: Web3Shim.utils.fromWei(gasPrice, "gwei"),
+      gas,
+      gasUnit: "gwei",
+      value: Web3Shim.utils.fromWei(value, "ether"),
+      cost,
+      totalUnit: "ETH",
```

Comment:
Should this just be named `unit`? Or maybe `unitType`? Why "total"?

Reply:
Oh I see how it is used, should there be separate unit types sent for "total cost" and "value sent"? Is it likely they will ever be different?

Reply:
Or we could consider something like the following pattern:
```
cost: {
  value: 2444,
  unit: "ether"
}
```

II'm not sure if that is overkill though.

Reply:
I did it like that on the first commit, but then I changed my mind as balance, value and cost were using the same unit (which should be the same for tezos, with TEZ as the main unit and mutez for the fee).

In retrospective, `totalUnit` is not a good name, `mainUnit` or `displayUnit` would probably be better.

Reply:
There's something else I could do similar to what you suggest, but it might be a bit overkill.

Since some of the values I return are already strings (as they don't need to be added up), and some are BNs, I could just always return
```
type BNValue = {
  value: BN,
  displayUnit: string,
  convertMethod: (val: BN) => string
}
```

That way I don't make assumptions of what the reporter needs and which values will need to be added up. It might look a bit cumbersome though.